main.go

```diff
@@ -32,54 +37,84 @@ func getConfig() config {
 	url := os.Getenv("JENKINS_BASE_URL")
 	githubSecret := os.Getenv("GITHUB_SECRET")
 	githubToken := os.Getenv("GITHUB_TOKEN")
+	integrationDirectory := os.Getenv("INTEGRATION_DIRECTORY")
+
+	// repos that require to be tested with specific integration branches
+	integrationBranchDependant :=
+		[]string{
+			"deployments",
+			"deviceadm",
+			"deviceauth",
+			"useradm",
+			"inventory",
+			"mender-api-gateway-docker",
+			"mender",
+		}
+
+	// if no env. variable is set, this is the default repo watch list
+	defaultWatchRepositories :=
+		[]string{
+			"deployments",
+			"deviceadm",
+			"deviceauth",
+			"inventory",
+			"useradm",
+			"integration",
+			"mender",
+			"mender-qa",
+			"mender-artifact",
+			"meta-mender",
+			"mender-api-gateway-docker"}
 
-	defaultWatchRepositories := "deployments,deviceadm,deviceauth,inventory,useradm,integration,mender,meta-mender"
 	watchRepositories := os.Getenv("WATCH_REPOS")
 
 	if len(watchRepositories) == 0 {
-		repositoryWatchList = strings.Split(defaultWatchRepositories, ",")
+		repositoryWatchList = defaultWatchRepositories
 	} else {
 		repositoryWatchList = strings.Split(watchRepositories, ",")
 	}
 
-	if username == "" {
+	switch {
+	case username == "":
 		panic("set JENKINS_USERNAME")
-	}
-
-	if password == "" {
+	case password == "":
 		panic("set JENKINS_PASSWORD")
-	}
-
-	if url == "" {
+	case url == "":
 		panic("set JENKINS_BASE_URL")
-	}
-
-	if githubSecret == "" {
+	case githubSecret == "":
 		panic("set GITHUB_SECRET")
-	}
-
-	if githubToken == "" {
+	case githubToken == "":
 		panic("set GITHUB_TOKEN")
+	case integrationDirectory == "":
+		panic("set INTEGRATION_DIRECTORY")
 	}
 
-	return config{username, password, url, []byte(githubSecret), githubToken, repositoryWatchList}
+	return config{username,
```

Comment:
tiny thing but use assignments with field names, if you add new field not at the end everything will get assigned to different attribute  and may even compile